ui/src/UI/Components/Ripple/Ripple.xaml.cs

```diff
@@ -0,0 +1,158 @@
+        private const int FrameCount = 45;
+
+        private Task animationTask;
+        private CancellationTokenSource animationTaskCts;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Ripple"/> class.
+        /// </summary>
+        public Ripple()
+        {
+            InitializeComponent();
+            SetAnimation();
+        }
+
+        /// <inheritdoc/>
+        public event PropertyChangedEventHandler PropertyChanged;
+
+        /// <summary>
+        /// Gets or sets a value indicating whether the component animation is running or not.
+        /// </summary>
+        public bool Running
+        {
+            get
+            {
+                return (bool)GetValue(RunningProperty);
+            }
+
+            set
+            {
+                SetValue(RunningProperty, value);
+                OnPropertyChanged("Running");
+            }
+        }
+
+        /// <summary>
+        /// Reacts to dynamic changes to properties within this control.
+        /// </summary>
+        /// <param name="propertyName">Name of the changed property.</param>
+        public virtual void OnPropertyChanged(string propertyName)
+        {
+            PropertyChangedEventHandler handler = this.PropertyChanged;
+            if (handler != null)
+            {
+                var e = new PropertyChangedEventArgs(propertyName);
+                handler(this, e);
+            }
+
+            if (propertyName == "Running")
+            {
+                SetAnimation();
+            }
+        }
+
+        private void SetAnimation()
+        {
+            if (Running == true)
+            {
+                if (animationTask != null && !animationTask.IsCompleted)
+                {
+                    // Animation is already running, no need to proceed
+                    return;
+                }
+
+                // Fade in the animation
+                Dispatcher.Invoke(() =>
+                {
+                    RippleImage.Opacity = 0;
+                    if (this.FindResource("FadeInAnimation") is Storyboard fadeInStoryboard)
+                    {
+                        BeginStoryboard(fadeInStoryboard);
+                    }
+                });
+
+                animationTaskCts = new CancellationTokenSource();
+
+                // Invoke the animation scroller
+                animationTask = new Task(() =>
+                    {
+                        var position = 0;
+                        while (!animationTaskCts.IsCancellationRequested)
+                        {
+                            var newPos = new Rect(position, 0, FrameWidth, FrameHeight);
+                            Dispatcher.Invoke(() =>
+                            {
+                                RipplePosition.Rect = newPos;
+                            });
+
+                            position += FrameWidth;
+                            Thread.Sleep(FrameTimeoutMs);
```

Comment:
Just one last thing here,  animationTaskCts.WaitHandle.WaitOne(TimeSpan.FromMilliseconds(FrameTimeoutMs)); can be used here to avoid using Thread.Sleep()